.github/workflows/build.yaml

```diff
@@ -86,14 +87,29 @@ jobs:
           ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_KEY }}
           ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.GPG_PASSPHRASE }}
         run: |
-          gradle -b galasa-managers-parent/build.gradle check publish \
+          set -o pipefail
+          gradle -b galasa-managers-parent/build.gradle check publish --info \
           -Dorg.gradle.jvmargs=-Xmx4096M \
           -PsourceMaven=https://development.galasa.dev/${{ env.ARGO_APP_BRANCH }}/maven-repo/extensions \
           -PcentralMaven=https://repo.maven.apache.org/maven2/ \
           -PtargetMaven=${{github.workspace}}/repo \
           -PjacocoEnabled=${{ inputs.jacocoEnabled }} \
-          -PisMainOrRelease=${{ inputs.isMainOrRelease }}
-
+          -PisMainOrRelease=${{ inputs.isMainOrRelease }} 2>&1 | tee build.log
+        
+      - name: Upload Gradle Build Log
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: gradle-build-log
+          path: build.log
+          
+      - name: Upload Jacoco Report
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: jacoco-report
+          path: ${{github.workspace}}/galasa-managers-parent/build/reports/jacoco/jacocoMerge

```

Comment:
My colleague suggested a way to do this to sweep up all `.html` reports for each project which I think is more effective than using the jacocoMerge. Can you please update this to:
```
      - name: Upload Jacoco Reports as artifact
        if: failure()
        uses: actions/upload-artifact@v3
        with:
          name: unit-test-reports
          path: /home/runner/work/managers/managers/galasa-managers-parent/**/**/build/reports/**/*.html
          retention-days: 7
          if-no-files-found: ignore
```

Reply:
done